File: test-suite/src/test/java/io/micronaut/kafka/docs/quickstart/ProductClient.java
package io.micronaut.kafka.docs.quickstart;

// tag::imports[]
import io.micronaut.configuration.kafka.annotation.KafkaClient;
import io.micronaut.configuration.kafka.annotation.KafkaKey;
import io.micronaut.configuration.kafka.annotation.Topic;
import io.micronaut.context.annotation.Requires;
// end::imports[]

@Requires(property = "spec.name", value = "QuickstartTest")
// tag::clazz[]
@KafkaClient // <1>
public interface ProductClient {

    @Topic("my-products") // <2>
    void sendProduct(@KafkaKey String brand, String name); // <3>

    void sendProduct(@Topic String topic, @KafkaKey String brand, String name); // <4>
}
// end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/quickstart/ProductClient.kt
package io.micronaut.kafka.docs.quickstart

// tag::imports[]
import io.micronaut.configuration.kafka.annotation.KafkaClient
import io.micronaut.configuration.kafka.annotation.KafkaKey
import io.micronaut.configuration.kafka.annotation.Topic
import io.micronaut.context.annotation.Requires
// end::imports[]

@Requires(property = "spec.name", value = "QuickstartTest")
// tag::clazz[]
@KafkaClient // <1>
interface ProductClient {
    @Topic("my-products")  // <2>
    fun sendProduct(@KafkaKey brand: String, name: String) // <3>

    fun sendProduct(@Topic topic: String, @KafkaKey brand: String, name: String) // <4>
}
// end::clazz[]
